### PR TITLE
Rename ref to rev in version context configuration

### PR DESCRIPTION
**Pull request by asdine**
As asked by @felixfbecker in this [comment](https://docs.google.com/document/d/1Zfxl13iJzexsOsQQncmnR1gISFrQ6MhjsoDKqEqOX-4/edit?disco=AAAAGhD76w8), the "ref" field in the version context configuration has been renamed to "rev".
@attfarhan Is there anything to change on frontend side?

### CODE REVIEW

File: cmd/frontend/graphqlbackend/search.go [hunk 1]
```diff
@@ -680,7 +680,7 @@ func resolveRepositories(ctx context.Context, op resolveRepoOp) (repoRevisions,
 			for _, vcRepoRef := range versionContext.Revisions {
 				if vcRepoRef.Repo == string(repo.Name) {
 					repoRev.Repo = repo
-					revs = append(revs, search.RevisionSpecifier{RevSpec: vcRepoRef.Ref})
+					revs = append(revs, search.RevisionSpecifier{RevSpec: vcRepoRef.Rev})
```
Comment (felixfbecker) [inline cmd/frontend/graphqlbackend/search.go:683]: Should the variable also be renamed?
```suggestion
					revs = append(revs, search.RevisionSpecifier{RevSpec: vcRepoRev.Rev})
```

Comment (asdine) [reply cmd/frontend/graphqlbackend/search.go:683]: Good catch 👍

**Comment by attfarhan** [comment]
@asdine you'll need to update these tests:

- https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4ab650e5a239303ff8537216b11b8fffeba4d785/-/blob/web/src/search/index.test.tsx#L53-62

- https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4ab650e5a239303ff8537216b11b8fffeba4d785/-/blob/web/src/nav/VersionContextDropdown.story.tsx#L26-31

**Comment by attfarhan** [comment]
@asdine also a couple places in the docs: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/doc/user/search/index.md#L106:16-122:2

Do we plan to get this into the release? If so, also will need to change it in the blog post and ensure we cherry-pick the commit onto the 3.16 branch

**Comment by asdine** [comment]
All right, I'll create a PR for the documentation change and cherry pick both commits onto the 3.16 branch

**Comment by attfarhan** [comment]
cc @uwedeportivo -- @asdine you should let the release captain (Uwe) know and get approval before cherry-picking

**Comment by uwedeportivo** [comment]
@asdine LGTM, go ahead and cherry-pick

**Pull request merged**